Theme icons, take 1

LeaVerou opened the pull request:
This is a prototype of theme icons based on @lindsaym-fa’s sketch: 

![image](https://github.com/user-attachments/assets/912a81b1-e94b-443c-934c-7754466c7f18)

> Current thinking is something like this —
> 
> - Overall background uses the default surface color. Maybe we also outline this with the surface border color.
> - Text sample uses the heading font
> - Swatches show the loud fills for brand, success, warning, and danger
> - Brand swatch is larger and has icon that uses the corresponding “on” color
> - Alternatively, we could focus in on the brand color and show loud, normal, and quiet fills as swatches
> - Second row includes a small input with placeholder text and a small brand button
> 
> That seems like it would be the minimum to get an impression of fonts, colors, rounding, borders, tactility, etc.
> 
> Open to suggestions or other explorations!

Screenshot:
<img width="1143" alt="image" src="https://github.com/user-attachments/assets/73123151-bfe7-4cd0-9f79-24a3bd216947" />

With `--wa-color-surface-default`:
<img width="1139" alt="image" src="https://github.com/user-attachments/assets/1813d0b9-f4e5-4aa0-8bf3-ac84ce3e141c" />

The icons are using the actual CSS files (rendered in shadow DOM) so they can easily work with custom themes.
I did not include the icon on the brand swatch since it will just be noise until we can have different icons per theme (but I’m not opposed to adding them now if you feel that's best).

I feel they look a bit too noisy all together, but it gives us a starting point we can improve on. Of course it goes without saying that if you have the cycles @lindsaym-fa feel free to go in this PR and tweak them to your heart's content.

LeaVerou committed:
Theme icons, take 1

lindsaym-fa committed:
Change balance of color swatches

lindsaym-fa reviewed (approved):
I think this works well overall! I did tweak the color swatches a bit to reduce their height and the whitespace between. To me, this helps the swatches draw less attention, which might have been contributing to the optical noise.

<img width="1026" alt="Screenshot 2025-03-26 at 11 33 52 AM" src="https://github.com/user-attachments/assets/1146df3f-f707-482a-97dd-d10700361a65" />
 
I did notice that the sample text doesn't always render with the theme's heading font family, like in the screenshot above. Note the Awesome theme icon isn't rendering Quicksand, Brutalist isn't rendering IBM Plex Sans, Playful isn't rendering Fredoka, etc. Approved in the meantime if we want to circle back to fix that later.

LeaVerou commented:
> I did tweak the color swatches a bit to reduce their height and the whitespace between. To me, this helps the swatches draw less attention, which might have been contributing to the optical noise.

I love it! Much, much better.

> I did notice that the sample text doesn't always render with the theme's heading font family, like in the screenshot above. Note the Awesome theme icon isn't rendering Quicksand, Brutalist isn't rendering IBM Plex Sans, Playful isn't rendering Fredoka, etc. 

It did feel that something was off about the typography, glad you were able to identify it! It seems to me that in all of these cases, the issue is that a web font is not loading, can you confirm?

LeaVerou committed:
Fix web fonts in theme icons

Instead of raw DSD, use a component that pulls in a child template and then goes over the CSS and extracts font-related rules into the document, just once per rule.
This also fixes theme icons in Vue.

LeaVerou committed:
Workaround for dark mode
